testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java

```diff
@@ -4164,6 +4164,77 @@ void checkPeriodPredicate(Checker checker) {
     sql(sql).ok(expected);
   }
 
+  @Test void testTableFunction() {
+    final String sql = "select * from table(score(table orders))";
+    final String expected = "SELECT *\n"
+        + "FROM TABLE(`SCORE`((TABLE `ORDERS`)))";
+    sql(sql).ok(expected);
+  }
+
+  @Test void testTableFunctionWithPartitionKey() {
+    // test one partition key for input table
+    final String sql = "select * from table(topn(table orders partition by productid, 3))";
+    final String expected = "SELECT *\n"
+        + "FROM TABLE(`TOPN`(((TABLE `ORDERS`) PARTITION BY `PRODUCTID`), 3))";
+    sql(sql).ok(expected);
+  }
+
+  @Test void testTableFunctionWithMultiplePartitionKeys() {
+    // test multiple partition keys for input table
+    final String sql =
+        "select * from table(topn(table orders partition by (orderId, productid), 3))";
+    final String expected = "SELECT *\n"
+        + "FROM TABLE(`TOPN`(((TABLE `ORDERS`) PARTITION BY `ORDERID`, `PRODUCTID`), 3))";
+    sql(sql).ok(expected);
+  }
```

Comment:
We need some tests for multiple partition/ordering keys.

Reply:
> We need some tests for multiple partition/ordering keys.

There are already exist cases to cover these, for example, SqlParserTest#testTableFunctionWithMultiplePartitionKeys, SqlParserTest#testTableFunctionWithMultipleOrderKeys


> We need some tests for multiple table params.

Based on SQL standard, a table function at most has one input table with row semantics, there are already exist cases to cover these based on MockSqlOperatorTable#InvalidTableFunction.

I add some cases to cover a table function with multiple input tables with set semantics based on MockSqlOperatorTable#SimilarlityTableFunction.